sdk/logical/logical.go

```diff
@@ -104,6 +108,13 @@ type BackendConfig struct {
 	Config map[string]string
 }
 
+// InitializationRequest is provided to backend.Initialize()
+type InitializationRequest struct {
```

Comment:
Why make this a new type? `logical.Request` has Storage as well. Making this a new type means over the plugin interface we have to create a new proto.

Reply:
I did it this way at @briankassouf 's suggestion, since `logical.Request` has a bunch of stuff on it that is irrelevant to what we are trying to do here.  I made the associated changes to the `backend.proto` as well.  It was my understanding from a discussion with @chrishoffman that since all of the plugins inside of the vault repo use framework.Backend, we wouldn't have to change anything else other than what I have already done.

Having said that, the current design does bother me, since when we `Initialize()` from the client side, there is nothing to pass in:

https://github.com/hashicorp/vault/blob/aws-upgrade-role-entries/vault/mount.go#L1121

and we only instantiate an `InitializationRequest` on the server side of GRPC, by pulling the `Storage` out of the brokered client:

https://github.com/hashicorp/vault/blob/aws-upgrade-role-entries/sdk/plugin/grpc_backend_server.go#L92-L94

That doesn't seem very pretty to me, but I'm not sure what a better design would look like.